File: solutions/2026-02-01/solution.java
// Validate IP Address
// Difficulty: Medium
// URL: https://leetcode.com/problems/validate-ip-address/

public boolean validIPAddress(String s) {
  public boolean validPart(String part) {
    if (part.length() > 3 || (part.length() > 1 && part.charAt(0) == '0')) {
      return false;
    }
    if (!part.matches("[0-9]+")) {
      return false;
    }
    return true;
  }
  String[] parts = s.split(".");
  if (parts.length != 4) {
    return false;
  }
  for (String part : parts) {
    if (!validPart(part)) {
      return false;
    }
  }
  return true;
}

File: solutions/2026-02-01/solution.py
# Validate IP Address
# Difficulty: Medium
# URL: https://leetcode.com/problems/validate-ip-address/

def validIPAddress(s: str) -> bool:
  def valid_part(part: str) -> bool:
    if len(part) > 3 or (len(part) > 1 and part[0] == '0'):
      return False
    if not part.isdigit():
      return False
    return True
  parts = s.split('.')
  if len(parts) != 4:
    return False
  for part in parts:
    if not valid_part(part):
      return False
  return True